include/lbann/data_readers/data_reader.hpp

```diff
@@ -531,11 +531,263 @@ class generic_data_reader : public lbann_image_preprocessor {
   void use_unused_index_set();
 
   /** \brief Given directory to store checkpoint files, write state to file and add to number of bytes written */
-  bool saveToCheckpointShared(persist& p, const char *name) const;
+  bool save_to_checkpoint_shared(persist& p, const char *name);
 
   /** \brief Given directory to store checkpoint files, read state from file and add to number of bytes read */
-  bool loadFromCheckpointShared(persist& p, const char *name);
+  bool load_from_checkpoint_shared(persist& p, const char *name);
+
+  struct packing_header {
+    uint64_t mini_batch_size;
+    uint64_t current_pos;
+    uint64_t current_mini_batch_idx;
+    uint64_t data_size;
+    std::vector<int> shuffled_indices;
+    uint64_t unused_data_size;
+    std::vector<int> unused_indices;
+    uint64_t stride_to_last_mini_batch;
+    uint64_t stride_to_next_mini_batch;
+    uint64_t base_offset;
+    uint64_t model_offset;
+    uint64_t sample_stride;
+    uint64_t iteration_stride;
+    uint64_t loaded_mini_batch_idx;
+    uint64_t reset_mini_batch_index;
+    uint64_t last_mini_batch_size;
+    uint64_t num_iterations_per_epoch;
+    uint64_t global_mini_batch_size;
+    uint64_t global_last_mini_batch_size;
+    uint64_t num_parallel_readers;
+    uint64_t model_rank;
```

Comment:
I added a new field to the data reader in my last string of fixes. The field relates to the world model remained and needs to be stored.

Reply:
added parameter 'world_master_mini_batch_adjustment' to save/load, and am merging now.